front/lib/api/assistant/conversation.ts

```diff
@@ -1120,270 +1121,302 @@ export async function* editUserMessage(
-      configurationId: agentMessage.configuration.sId,
-      messageId: agentMessage.sId,
-      message: agentMessage,
+      messageId: userMessage.sId,
+      message: userMessage,
     };
-  }
 
-  const eventStreamGenerators = agentMessages.map((agentMessage, i) => {
-    // We stitch the conversation to add the user message and only that agent message
-    // so that it can be used to prompt the agent.
-    const eventStream = runAgent(
-      auth,
-      agentMessage.configuration,
-      {
-        ...conversation,
-        content: [...conversation.content, [userMessage], [agentMessage]],
-      },
-      userMessage,
-      agentMessage
-    );
+    for (let i = 0; i < agentMessages.length; i++) {
+      const agentMessage = agentMessages[i];
 
-    return streamRunAgentEvents(
-      auth,
-      eventStream,
-      agentMessages[i],
-      agentMessageRows[i]
-    );
-  });
+      yield {
+        type: "agent_message_new",
+        created: Date.now(),
+        configurationId: agentMessage.configuration.sId,
+        messageId: agentMessage.sId,
+        message: agentMessage,
+      };
+    }
 
-  const eventStreamsPromises = eventStreamGenerators.map((gen) => gen.next());
-  while (eventStreamsPromises.length > 0) {
-    const winner = await Promise.race(
-      eventStreamsPromises.map(async (p, i) => {
-        return { v: await p, offset: i };
-      })
-    );
-    if (winner.v.done) {
-      eventStreamGenerators.splice(winner.offset, 1);
-      eventStreamsPromises.splice(winner.offset, 1);
+    const eventStreamGenerators = agentMessages.map((agentMessage, i) => {
+      // We stitch the conversation to add the user message and only that agent message
+      // so that it can be used to prompt the agent.
+      const eventStream = runAgent(
+        auth,
+        agentMessage.configuration,
+        {
+          ...conversation,
+          content: [...conversation.content, [userMessage], [agentMessage]],
+        },
+        userMessage,
+        agentMessage
+      );
+
+      return streamRunAgentEvents(
+        auth,
+        eventStream,
+        agentMessages[i],
+        agentMessageRows[i]
+      );
+    });
+
+    const eventStreamsPromises = eventStreamGenerators.map((gen) => gen.next());
+    while (eventStreamsPromises.length > 0) {
+      const winner = await Promise.race(
+        eventStreamsPromises.map(async (p, i) => {
+          return { v: await p, offset: i };
+        })
+      );
+      if (winner.v.done) {
+        eventStreamGenerators.splice(winner.offset, 1);
+        eventStreamsPromises.splice(winner.offset, 1);
+      } else {
+        eventStreamsPromises[winner.offset] =
+          eventStreamGenerators[winner.offset].next();
+        yield winner.v.value;
+      }
+    }
+  } catch (e) {
+    if (e instanceof UserMessageError) {
+      yield {
+        type: "user_message_error",
+        created: Date.now(),
+        error: {
+          code: "edition_unsupported",
```

Comment:
I think `edit_invalid_error` is marginally better. Edit is supported it's just that the request was invalid